task2-3/src/services/entityDataMapperService.ts

```diff
@@ -1,4 +1,4 @@
-export default abstract class EntityDataMapperService {
+export default class EntityDataMapperService {
```

Comment:
mappers are usually stored in a separate location. Example: _mappers_ folder
They are used to transfer data from the database into objects that your domain model can work with and process

So each of your mappers should have at least two methods:

- **toPersistance** - converts your object(User, UserGroup etc.) into a form that can be stored in the database
- **toDomain** - converts your DB data to your domain object(User, UserGroup etc.) 

You could benefit a lot if you read this article and use this patterns in you project
https://khalilstemmler.com/articles/typescript-domain-driven-design/repository-dto-mapper/